src/main/java/org/apache/cassandra/metrics/MetricsRegistry.java

```diff
@@ -654,7 +654,11 @@ public JmxTimer(String url, long interval) {
         public void update(JsonObject obj) {
             // TODO: this is not atomic.
             super.update(obj.getJsonObject("meter"));
-            histogram = new Histogram(obj.getJsonObject("hist"));
+            if (obj.containsKey("hist")) {
+                histogram = new Histogram(obj.getJsonObject("hist"));
+            } else if (obj.containsKey("buckets")) {
+                histogram = new Histogram(new EstimatedHistogram(obj));
+            }
```

Comment:
The logic in `JmxHistogram` has an additional `else` path:

```java
        @Override
        public void update(JsonObject obj) {
            if (obj.containsKey("hist")) {
                obj = obj.getJsonObject("hist");
            }
            if (obj.containsKey("buckets")) {
                histogram = new Histogram(new EstimatedHistogram(obj));
            } else {
                histogram = new Histogram(obj);
            }
        }
```

Which one of the two is correct?

Reply:
it's mutually exclusive (it's either hist or buckets) but I'm closing this PR because I will take a different path